acceptance/acceptance_test.go

```diff
@@ -193,3 +200,86 @@ func createdArchiveIsEmpty(ctx context.Context) (context.Context, error) {
 
 	return ctx, nil
 }
+
+func createFiles(ctx context.Context, files *godog.Table) (context.Context, error) {
+	if files == nil {
+		return ctx, nil
+	}
+	ts, err := getTestState(ctx)
+	if err != nil {
+		return ctx, fmt.Errorf("no test state: %w", err)
+	}
+
+	sourceDir := ts.sourceDir()
+
+	for _, row := range files.Rows[1:] {
+		path := row.Cells[0].Value
+		content := row.Cells[1].Value
+
+		fpath := filepath.Join(sourceDir, path)
+
+		if err := os.MkdirAll(filepath.Dir(fpath), 0700); err != nil {
+			return ctx, err
+		}
+
+		if err := os.WriteFile(fpath, []byte(content), 0400); err != nil {
+			return ctx, err
+		}
+	}
+
+	return ctx, nil
+}
+
+func artifactContains(ctx context.Context, result string, files *godog.Table) (context.Context, error) {
+	ts, err := getTestState(ctx)
+	if err != nil {
+		return ctx, fmt.Errorf("no test state: %w", err)
+	}
+
+	archiveUri, err := os.ReadFile(filepath.Join(ts.resultsDir(), result))
+	if err != nil {
+		return ctx, fmt.Errorf("reading result file: %w", err)
+	}
+
+	// Set up the file paths as they will be seen within the container.
+	mountedTS := ts.forMount("/data")
+	storePath := mountedTS.artifactsDir()
+	restoredPath := mountedTS.restoredDir()
+
+	binds := []string{
+		// TODO: The ":Z" option is required on Linux systems because of to selinux. This might not
+		// work on a mac, for example.
+		fmt.Sprintf("%s:%s:Z", ts.contextDir, mountedTS.contextDir),
+	}
+
+	cmd := []string{
+		"use",
+		"--store",
+		storePath,
+		fmt.Sprintf("%s=%s", archiveUri, restoredPath),
+	}
+
+	if err := runContainer(ctx, cmd, binds); err != nil {
+		return ctx, fmt.Errorf("using artifact: %w", err)
+	}
+
+	restoredDir := ts.restoredDir()
+
+	for _, row := range files.Rows[1:] {
+		path := row.Cells[0].Value
+		content := row.Cells[1].Value
+
+		fpath := filepath.Join(restoredDir, path)
+
+		bytes, err := os.ReadFile(fpath)
+		if err != nil {
+			return ctx, err
+		}
+
+		if got, expected := string(bytes), content; got != expected {
+			return ctx, fmt.Errorf("file at path %q differs when restored\n  expected:\n%s\n  got:n%s", fpath, expected, got)
```

Comment:
Check out how `restoredFileShouldMatchSource` does the comparison of file contents. It would be great to use that here as well. (Or to abstract that into a helper function.)